sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2223,3 +2223,29 @@ object TimeWindowing extends Rule[LogicalPlan] {
       }
   }
 }
+
+/**
+ * Replaces ordinal in 'order by' or 'group by' with unresolved UnresolvedOrdinal expression.
+ */
+class UnresolvedOrdinalSubstitution(conf: CatalystConf) extends Rule[LogicalPlan] {
```

Comment:
if we end up doing it this way, move this to its own file, and create an invididual test suite. 

analyzer file is getting too large.